packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -440,6 +440,7 @@ class Context {
       '-dTargetFile=$targetPath',
       '-dBuildMode=$buildMode',
       if (environment['FLAVOR'] != null) '-dFlavor=${environment['FLAVOR']}',
+      '-dConfiguration=${environment['CONFIGURATION']}',
```

Comment:
Is threre a reason this isn't similar to line 442; that is, if this is null, we would print`=null`, which I'm not sure is intended (if it is, probably deserves a comment)

Reply:
Yes, so `FLAVOR` is set by the Flutter CLI in Flutter/Generated.xcconfig when the `--flavor` is used, so may not always be present. `CONFIGURATION` is provided by Xcode/xcodebuild and should always be present. I'll add a comment explaining why `FLAVOR` may be null